clang/lib/CIR/Dialect/Transforms/TargetLowering/LowerFunction.h

```diff
@@ -0,0 +1,65 @@
+//===-- LowerFunction.h - Per-Function state for CIR lowering ---*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This class partially mimics clang/lib/CodeGen/CGFunctionInfo.h. The queries
+// are adapted to operate on the CIR dialect, however. And we only copy code
+// related to ABI-specific codegen.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_CIR_DIALECT_TRANSFORMS_TARGETLOWERING_LOWERFUNCTION_H
+#define LLVM_CLANG_LIB_CIR_DIALECT_TRANSFORMS_TARGETLOWERING_LOWERFUNCTION_H
+
+#include "CIRCXXABI.h"
+#include "LowerCall.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Support/LogicalResult.h"
+#include "clang/Basic/TargetInfo.h"
+#include "clang/CIR/Dialect/IR/CIRDialect.h"
+
+namespace mlir {
+namespace cir {
+
+/// The kind of evaluation to perform on values of a particular
+/// type.  Basically, is the code in CGExprScalar, CGExprComplex, or
+/// CGExprAgg?
+///
+/// TODO: should vectors maybe be split out into their own thing?
+enum TypeEvaluationKind { TEK_Scalar, TEK_Complex, TEK_Aggregate };
```

Comment:
This already exists in `clang/lib/CIR/CodeGen/CIRGenFunction.h`, please unify instead of duplicating. It's already too hacky the way is used, let's not make it worse.

Reply:
Should we isolate `TypeEvaluationKind` in a new header for use within CIR?

Reply:
Doesn't seem like this got fixed, unresolving